ObjectOrientedProgramming/Memory Ownership/unique_ptr/task.md

```diff
@@ -0,0 +1,50 @@
+C++ contains a tool that provides exclusive ownership semantics. [std::unique_ptr](https://en.cppreference.com/w/cpp/memory/unique_ptr) is a "smart" pointer that, like all other smart pointers, is located in the `<memory>` library (so don't forget to connect it), and is designed to manage the lifetime of dynamically allocated objects. The unique ownership model ensures that at any given time, only one `std::unique_ptr` instance owns a particular dynamically allocated object. When the owning `std::unique_ptr` is destroyed or explicitly reset, it ensures the deallocation of the associated memory.
```

Comment:
> is located in the `<memory>` library (so don't forget to connect it)

we do not explicitly mentioned header files so far in this course, so this info should be omitted